src/mesh/exodusII_io.C

```diff
@@ -286,7 +286,30 @@ void ExodusII_IO::read (const std::string & fname)
 
           // Assign extra integer IDs
           for (auto & id : extra_ids)
-            elem->set_extra_integer(id, std::round(elem_ids[id][elem->id()]));
+          {
+            Real v = elem_ids[id][elem->id()];
+            if (std::numeric_limits<Real>::digits > std::numeric_limits<dof_id_type>::digits)
+              elem->set_extra_integer(id, std::round(v));
+            else
+            {
+              if (v == (Real) DofObject::invalid_id)
+                elem->set_extra_integer(id, DofObject::invalid_id);
+              else
+              {
+                dof_id_type max_representation = 1;
```

Comment:
You might need to explicitly make this unsigned long to avoid that overflow warning with 32-bit dof_id_type?

Although it does seem like the compiler could be smarter here; you shouldn't get an overflow unless you're in the other branch of the if statement, which is using a constexpr argument so should be evaluable at compile time.

Reply:
I can do that.